packages/orchestration/src/facade.js

```diff
@@ -14,6 +14,19 @@ import { assertAllDefined } from '@agoric/internal';
  * @import {Chain, ChainInfo, CosmosChainInfo, IBCConnectionInfo, OrchestrationAccount, Orchestrator} from './types.js';
  */
 
+/**
+ * For a given guest passed to orchestrate(), return the host-side form.
+ *
+ * @template {(orc: Orchestrator, ctx: any, ...args: any[]) => Promise<any>} GF
+ * @typedef {GF extends (
+ *   orc: Orchestrator,
+ *   ctx: any,
+ *   ...args: infer GA
+ * ) => Promise<infer GR>
+ *   ? (...args: HostArgs<GA>) => Vow<GR>
+ *   : never} HostForGuest
```

Comment:
this is at the edge of my tsc-fu; an example in the form of a type test would help me.

Reply:
worthwhile. probably another PR.